azurerm/resource_arm_kusto_cluster_test.go

```diff
@@ -0,0 +1,242 @@
+					resource.TestCheckResourceAttr(resourceName, "tags.label", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.ENV", "prod"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMKustoCluster_sku(t *testing.T) {
+	resourceName := "azurerm_kusto_cluster.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(6)
+	preConfig := testAccAzureRMKustoCluster_basic(ri, rs, testLocation())
+	postConfig := testAccAzureRMKustoCluster_skuUpdate(ri, rs, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMKustoClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: preConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKustoClusterExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "sku.0.name", "Dev(No SLA)_Standard_D11_v2"),
+					resource.TestCheckResourceAttr(resourceName, "sku.0.capacity", "1"),
+				),
+			},
+			{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKustoClusterExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "sku.0.name", "Standard_D11_v2"),
+					resource.TestCheckResourceAttr(resourceName, "sku.0.capacity", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAzureRMKustoCluster_basic(rInt int, rs string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%s"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku {
+    name     = "Dev(No SLA)_Standard_D11_v2"
+    capacity = 1
+  }
+}
+`, rInt, location, rs)
+}
+
+func testAccAzureRMKustoCluster_withTags(rInt int, rs string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%s"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  
+  sku {
+    name     = "Dev(No SLA)_Standard_D11_v2"
+    capacity = 1
+  }
+
+  tags = {
+	label = "test"
+  }
+}
+`, rInt, location, rs)
+}
+
+func testAccAzureRMKustoCluster_withTagsUpdate(rInt int, rs string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_kusto_cluster" "test" {
+  name                = "acctestkc%s"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku {
+    name     = "Dev(No SLA)_Standard_D11_v2"
+    capacity = 1
+  }
```

Comment:
this test fails because there's missing Tags in this configuration:

```
------- Stdout: -------
=== RUN   TestAccAzureRMKustoCluster_withTags
=== PAUSE TestAccAzureRMKustoCluster_withTags
=== CONT  TestAccAzureRMKustoCluster_withTags
--- FAIL: TestAccAzureRMKustoCluster_withTags (1266.52s)
    testing.go:568: Step 1 error: Check failed: Check 2/4 error: azurerm_kusto_cluster.test: Attribute 'tags.%' expected "2", got "1"
FAIL
```